src/init.cpp

```diff
@@ -1589,6 +1589,9 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         node.indexes.insert(g_coin_stats_index.get());
     }
 
+    // Init all indexes
+    for (auto index : node.indexes) if (!index->Init()) return false;
```

Comment:
In commit "refactor: index, decouple 'Init' from 'Start'" (9f38c2ffba87ea07c0eeb9975ad4d84ae716f9aa)

Oh wow, it seems like this commit is basically reverting all the behavior changes of earlier commit a9e45f7926fab62ef0c5cea540386266235b2990 "init: start indexes after 'loadblk' thread finishes" that I was confused by and asking about in my comment on that commit.

I would definitely recommend moving this commit before the earlier a9e45f7926fab62ef0c5cea540386266235b2990 commit, and keeping `RegisterValidationInterface` and `BaseIndex::Init` code running on the main thread throughout the PR, instead of moving it from the main thread to the loadblk thread in the earlier commit, and then moving it back to the main thread again in this commit. That would be a lot less confusing.

Reply:
yeah, totally agree. Commits order ended up quite ugly after all the findings. Thanks.